File: src/main/java/com/example/edusence_studio/services/feedbacks/ProblemTagService.java
package com.example.edusence_studio.services.feedbacks;

import com.example.edusence_studio.models.feedbacks.ProblemTag;

import java.util.List;
import java.util.UUID;

public interface ProblemTagService {
    List<ProblemTag> getAllProblemTags();
    ProblemTag getProblemTagById(UUID id);
    ProblemTag getProblemTagByCode(String code);
    ProblemTag createProblemTag(String code, String description);
    ProblemTag updateProblemTag(UUID id, String code, String description);
    void deleteProblemTag(UUID id);
}

File: frontend/src/services/problemTagService.js
import api from './api';

export const problemTagService = {
  getAllProblemTags: async () => {
    const response = await api.get('/problem-tags');
    return response.data;
  },

  getProblemTag: async (id) => {
    const response = await api.get(`/problem-tags/${id}`);
    return response.data;
  },

  createProblemTag: async (code, description) => {
    const response = await api.post('/problem-tags', null, {
      params: { code, description },
    });
    return response.data;
  },

  updateProblemTag: async (id, code, description) => {
    const response = await api.put(`/problem-tags/${id}`, null, {
      params: { code, description },
    });
    return response.data;
  },

  deleteProblemTag: async (id) => {
    await api.delete(`/problem-tags/${id}`);
  },
};
